File: Leetcode/Cpp/Easy/Roman to Integer.cpp
// https://leetcode.com/problems/roman-to-integer/

class Solution {
public:
    int converter(char x){
        int y;
        switch(x){
            case 'I':
                y = 1;
                break;
            case 'V':
                y = 5;
                break;
            case 'X':
                y = 10;
                break;
            case 'L':
                y = 50;
                break;
            case 'C':
                y = 100;
                break;
            case 'D':
                y = 500;
                break;
            case 'M':
                y = 1000;
                break;
        }
        return y;
    }
    int romanToInt(string s) {
        int n = s.length(), finalans = 0;
        for(int i = 0; i<n; i++){
            char ss = s[i];
            finalans += converter(ss);
            if(i-1 >= 0){
                char s1 = s[i-1];
                if(s[i-1] == 'I' && s[i] == 'V')
                    finalans -= 2*(converter(s1));
                else if(s[i-1] == 'I' && s[i] == 'X')

                    finalans -= 2*(converter(s1));
                
                else if(s[i-1] == 'X' && s[i] == 'L')
                    finalans -= 2*(converter(s1));
                else if(s[i-1] == 'X' && s[i] == 'C')
                    finalans -= 2*(converter(s1));
                
                else if(s[i-1] == 'C' && s[i] == 'D')
                    finalans -= 2*(converter(s1));
                else if(s[i-1] == 'C' && s[i] == 'M')
                    finalans -= 2*(converter(s1));
                
                }
        
            }
        return finalans;
        }
};

File: Leetcode/Python/Easy/Roman to Integer.py
# https://leetcode.com/problems/roman-to-integer/

class Solution(object):

    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        self.s = s
        final_Val = 0
        currentVal = ""
        previousVal = ""
        putVal = lambda x: (conversion_Dic[currentVal] - 2*conversion_Dic[previousVal])
        conversion_Dic = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
        for i in range(len(self.s)):
            currentVal = self.s[i]
            if i != 0:
                previousVal = self.s[i-1]
            if currentVal == "V" and previousVal == "I":
                final_Val += putVal(final_Val)

            elif currentVal == "X" and previousVal == "I":
                final_Val += putVal(final_Val)

            elif currentVal == "L" and previousVal == "X":
                final_Val += putVal(final_Val)

            elif currentVal == "C" and previousVal == "X":
                final_Val += putVal(final_Val)

            elif currentVal == "D" and previousVal == "C":
                final_Val += putVal(final_Val)

            elif currentVal == "M" and previousVal == "C":
                final_Val += putVal(final_Val)

            else:
                final_Val +=conversion_Dic[self.s[i]]
        return final_Val